### DISCUSSION TITLE
Returning directories in search results

### DISCUSSION THREAD

**Discussion by dtremp007** [sharkdp/fd#1152] [Q&A] [answered]
Hi, I'm using fd with fzf, but I've noticed it only returns files, not directories. Is there a flag for this?

**Comment by tavianator** [comment, marked as answer]
`fd` prints files and directories by default:

```console
tavianator@graphene $ fd
a_dir/
a_dir/a_file
```

**Reply by dtremp007** [reply]
You're right. When I set up my fzf, I set --type to f. My bad. Thanks.

**Comment by tmccombs** [comment]
But you can use the --type option to specify only directories or only files

**Reply by dtremp007** [reply]
That helps. Thanks 👍

**Comment by flawrist** [comment]
The frontpage of FD says “use fd PATTERN instead of find -INAME”. Thing is I could specify multiple inames with the latter to achieve AND condition. For example, `-iname *dogs* -iname *books*` would find _Folder with books about dogs_. But if I launch ```fd dogs books``` then…
```
[fd error]: Search path 'books' is not a directory.
[fd error]: No valid search paths given.
```
I looked for a solution in the following issues: #315, #650, #714.

Any ideas?

**Reply by tavianator** [reply]
Currently `fd` doesn't support this (though see https://github.com/sharkdp/fd/pull/1139).  You can always do `fd dogs | grep -i books`

**Reply by flawrist** [reply]
Hmm… `fd --regex "(?:.*books)(?:.*dogs)"` seems to do the trick, but if you change the order of the words, i.e. `"(?:.*dogs)(?:.*books)"` it breaks down. And look-aheads such as `"(?=.*dogs)(?=.*books)"` are still not supported. So you have to call an external tool, indeed. For example, `fd … | rg --pcre2 "(?=.*dogs)(?=.*books)"`. But it makes me sad in the 21st century.

Speaking of external search tools, there is a Windows gem called [Everthing](https://voidtools.com/), which has a command-line support. To assist this case one has to launch `es.exe /ad dogs books`. Human-readable syntax that soothes the soul.

**Reply by tmccombs** [reply]
Fwiw that pull request is fairly close to getting merged.

**Reply by sharkdp** [reply]
This is possible since `fd` 8.6:
```
fd dogs --and books
```
